src/lib.rs

```diff
@@ -96,70 +97,45 @@ pub fn load_and_index_pois(
     })?;
     let admins_geofinder = admins.into_iter().collect();
 
-    let bbox_filter = args
-        .bounding_box
-        .as_ref()
-        .map(|b| {
-            format!(
-                "WHERE ST_MakeEnvelope({}, 4326) && st_transform(geometry, 4326)",
-                b
-            )
-        })
-        .unwrap_or_else(|| "".into());
+    use pg_poi_query::{POIsQuery, TableQuery};
 
-    let query = format!(
-        "
-        SELECT
-            id,
-            lon,
-            lat,
-            class,
-            name,
-            tags,
-            subclass,
-            mapping_key,
-            poi_display_weight(name, subclass, mapping_key, tags)::float as weight
-        FROM (
-            SELECT
-                geometry,
-                global_id AS id,
-                st_x(st_transform(geometry, 4326)) AS lon,
-                st_y(st_transform(geometry, 4326)) AS lat,
-                class,
-                name,
-                mapping_key,
-                subclass,
-                tags
-            FROM all_pois(14)
-            UNION ALL
-            SELECT
-                geometry,
-                global_id_from_imposm(osm_id) AS id,
-                st_x(st_transform(geometry, 4326)) AS lon,
-                st_y(st_transform(geometry, 4326)) AS lat,
-                'aerodrome' AS class,
-                name,
-                'aerodrome' AS mapping_key,
-                'airport' AS subclass,
-                tags
-            FROM osm_aerodrome_label_point
-            UNION ALL
-            SELECT
-                geometry,
-                global_id_from_imposm(osm_id) AS id,
-                st_x(st_transform(geometry, 4326)) AS lon,
-                st_y(st_transform(geometry, 4326)) AS lat,
-                'locality' AS class,
-                name,
-                'locality' AS mapping_key,
-                'hamlet' AS subclass,
-                tags
-            FROM osm_city_point
-                WHERE name <> '' AND place='hamlet'
-        ) AS unionall
-        {}",
-        bbox_filter
-    );
+    let mut query = POIsQuery::new()
+        .with_table(TableQuery::new("all_pois(14)").id_column("global_id"))
+        .with_table(
+            TableQuery::new("osm_aerodrome_label_point")
+                .override_class("aerodrome")
+                .override_subclass("airport"),
+        )
+        .with_table(
+            TableQuery::new("osm_city_point")
+                .override_class("locality")
+                .override_subclass("hamlet")
+                .filter("name <> '' AND place='hamlet'"),
+        )
+        .with_table(
+            TableQuery::new("osm_water_lakeline")
+                .override_class("water")
+                .override_subclass("lake"),
+        )
+        .with_table(
+            TableQuery::new("osm_water_point")
+                .override_class("water")
+                .override_subclass("river"),
```

Comment:
Is there a more specific value we could use, instead of hardcoding "river" ?

Reply:
And this default is actually quite bad, I'll change it to `water` too if we don't find any better.

We could attempt to compute a subclass like we do for POIs for the waters with an amenity or landuse tag.

Reply:
yes using 'water' both for class and subclass is probably good enough for now, as the value of the `water=` tag does not seem to be imported in the database.